velox/functions/sparksql/ArrayGetFunction.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/sparksql/ArrayGetFunction.h"
+#include "velox/functions/lib/SubscriptUtil.h"
+
+namespace facebook::velox::functions::sparksql {
+
+namespace {
+
+/// get(array, index) -> array[index]
+///
+/// - allows negative indices for arrays (returns NULL if index < 0).
+/// - allows out of bounds accesses for arrays (returns NULL if out of
+///    bounds).
+/// - index starts at 0 for arrays.
+class ArrayGetFunction : public SubscriptImpl<
+                             /* allowNegativeIndices */ true,
+                             /* nullOnNegativeIndices */ true,
+                             /* allowOutOfBound */ true,
+                             /* indexStartsAtOne */ false> {
+ public:
+  explicit ArrayGetFunction(bool allowCaching) : SubscriptImpl(allowCaching) {}
+};
+
+std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
+  std::vector<std::shared_ptr<exec::FunctionSignature>> signatures;
+
+  // array(T), integer -> T
+  signatures.push_back(exec::FunctionSignatureBuilder()
```

Comment:
https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/ArrayConstructor.cpp#L108

Please use this style to initialize the vector

Reply:
ok.